EpicLoot/src/Magic/MagicItemEffects/MultiShot.cs

```diff
@@ -36,8 +44,50 @@ public static void Attack_FireProjectileBurst_Prefix(Attack __instance)
                     __instance.m_projectileAccuracy = 2;
                 }
 
-                __instance.m_projectiles = 2;
+                if (__instance.m_projectileBursts != 1)
+                {
+                    __instance.m_projectiles = 2;
+                }
+                else
+                {
+                    __instance.m_projectiles *= 2;
+                }
+            }
+        }
+    }
+    
+    /// <summary>
+    /// Patch to remove thrice ammo when using TripleShot
+    /// </summary>
+    [HarmonyPatch(typeof(Attack), nameof(Attack.UseAmmo))]
+    public static class UseAmmoTranspilerPatch
+    {
+        static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            var code = new List<CodeInstruction>(instructions);
+
+            
+            var removeItemMethod = AccessTools.Method(typeof(Inventory), nameof(Inventory.RemoveItem), new Type[] { typeof(ItemDrop.ItemData), typeof(int) });
+            
+            for (int i = 0; i < code.Count; i++)
+            {
+                if (code[i].Calls(removeItemMethod))
+                {
+                    code[i] = new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(UseAmmoTranspilerPatch), nameof(CustomRemoveItem)));
+                }
+            }
+            
+            return code.AsEnumerable();
+        }
+        
+        public static bool CustomRemoveItem(Inventory inventory, ItemDrop.ItemData item, int amount)
+        {
+            if (MultiShot.isTripleShotActive)
```

Comment:
I don't see where this value gets set. Can this check be done with ``player.HasActiveMagicEffect(MagicEffectType.TripleBowShot)``? There may be a way to find the owner of the inventory and check if it is the local player. Potentially I can see an issue if this code runs for another mod where the inventory does not belong to the player, removing extra ammo it should not.

Reply:
I added a place for isTripleShotActive to be set and unset. I couldn't find a way to access player via the Inventory class.

https://github.com/OrianaVenture/Randy_Vapok_ValheimMods/pull/10/commits/1cdb6bfcbe56afdd63f7b8e75870347224bb19d6